README.md

```diff
@@ -212,36 +255,116 @@ $limit = $parameters->getLimit(100); // 100
 
-You can transform caught exceptions into JSON-API error documents using the `Tobscure\JsonApi\ErrorHandler` class. You must register the appropriate `Tobscure\JsonApi\Exception\Handler\ExceptionHandlerInterface` instances.
+A couple of implementations are already provided:
+
+* `Tobscure\JsonApi\Error\InternalServerErrorResponse` for a generic 500 Internal Server Error response.
+* `Tobscure\JsonApi\Error\InvalidParameterErrorResponse` for an error response corresponding to an `InvalidParameterException`.
+
+A Document containing the errors from an error response can be created using the `fromErrorResponse()` constructor:
 
 ```php
+use Tobscure\JsonApi\Document;
+use Tobscure\JsonApi\Error\InternalServerErrorResponse;
+
+$response = new InternalServerErrorResponse;
+
+$document = Document::fromErrorResponse($response);
+
+http_response_code($response->getStatusCode());
+header('Content-Type: ' . $document::MEDIA_TYPE);
+echo $document;
+```
+
+In order to translate a caught `Exception` into a JSON-API error response, you should implement the `ErrorResponseInterface` for each type of `Exception` you wish to handle:
+
+```php
+use Tobscure\JsonApi\Error;
+use Tobscure\JsonApi\Error\ErrorResponseInterface;
+
+class MyCustomErrorResponse implements ErrorResponseInterface
+{
+    protected $e;
+
+    public function __construct(MyCustomException $e)
+    {
+        $this->e = $e;
+    }
+
+    public function getStatusCode()
+    {
+        return 400;
+    }
+
+    public function getErrors()
+    {
+        $error = new Error;
+
+        $error->setId('1');
+        $error->setAboutLink('url');
+        $error->setMeta(['key' => 'value']);
+        $error->setStatusCode(400);
+        $error->setErrorCode('my-custom-error-code');
+        $error->setTitle('My Custom Error');
+        $error->setDetail('You dun goofed!');
+        $error->setSourcePointer('/data/attributes/custom');
+        $error->setSourceParameter('include');
+
+        return [$error];
+    }
+}
+```
+
+You can then instantiate the correct error response according to the type of `Exception` that has been caught. The `Tobscure\JsonApi\Error\DefaultErrorResponse` class can be used to automatically handle generation of an `InvalidParameterErrorResponse` and falling back to an `InternalServerErrorResponse`.
+
+```php
+use Tobscure\JsonApi\Document;
+use Tobscure\JsonApi\Error\DefaultErrorResponse;
+
 try {
     // API handling code
 } catch (Exception $e) {
-    $errors = new ErrorHandler;
-
-    $errors->registerHandler(new InvalidParameterExceptionHandler);
-    $errors->registerHandler(new FallbackExceptionHandler);
+    switch (true) {
+        case $e instanceof MyCustomException:
+            $response = new MyCustomErrorResponse($e);
+            break;
 
-    $response = $errors->handle($e);
+        default:
+            $response = DefaultErrorResponse::forException($e);
```

Comment:
`DefaultErrorResponse::forException($e)` is the best naming I could come up with. `DefaultErrorResponse` would be abstract, of course. Does this make sense?